yagi/src/main/java/org/raml/yagi/framework/grammar/rule/ErrorNodeFactory.java

```diff
@@ -74,7 +74,11 @@ public static Node createRequiredValueNotFound(Node node, Rule keyRule)
 
     public static Node createInvalidType(Node node, NodeType type)
     {
-        return new ErrorNode("Invalid type " + node.getType() + ", expected " + type);
+        return new ErrorNode("Invalid type " + node.getType() + ", expected " + type + getFieldMessageIfPresent(node));
+    }
+
+    private static String getFieldMessageIfPresent(Node node) {
+        return node.getParent() != null ? " for " + node.getParent().getChildren().get(0).toString() : "";
```

Comment:
is there any possibility that parent has no children?

Reply:
No